subsys/net/l2/openthread/openthread.c

```diff
@@ -246,13 +259,55 @@ static void openthread_process(void *context, void *arg2, void *arg3)
 {
 	struct openthread_context *ot_context = context;
 
+		// not restarting the joiner before a changing delay has passed
+		#define JOINER_DELAY_INITIAL 2000
+		#define JOINER_DELAY_MAX 32000
+		// if no join in 10 minutes, reboot with better luck
+		#define JOINER_REBOOT_TIME K_MINUTES(10)
+		s64_t joiner_delay_ms = JOINER_DELAY_INITIAL;
+		s64_t reboot_ref_ms = k_uptime_get();
+
+
 	while (1) {
 		while (otTaskletsArePending(ot_context->instance)) {
 			otTaskletsProcess(ot_context->instance);
 		}
 
 		otSysProcessDrivers(ot_context->instance);
 
+		// restart joiner if
+		// - in autostart mode and join hasn't succeeded yet
+		// - joiner is back to idle
+		if (joiner_autostart_started && !joiner_successful
+			&& otJoinerGetState(ot_context->instance)
+			== OT_JOINER_STATE_IDLE) {
+			if (k_uptime_get() - reboot_ref_ms >
+				JOINER_REBOOT_TIME) {
+				// printk doesn't seem to do the job
+				printf("Joining failed for %d seconds..."
```

Comment:
Please use logging macros

Reply:
Logging will not show up on UART, as the system is rebooting immediately

Reply:
Ok, please change this to `printk()` as that is using less resources.

Reply:
Printk doesn't work in this case. A explanation fr that might be in the discussion in https://github.com/zephyrproject-rtos/zephyr/pull/25627